src/hotspot/share/oops/arrayOop.hpp

```diff
@@ -71,6 +70,11 @@ class arrayOopDesc : public oopDesc {
   // aligned 0 mod 8.  The typeArrayOop itself must be aligned at least this
   // strongly.
   static bool element_type_should_be_aligned(BasicType type) {
+#ifdef _LP64
+    if (type == T_OBJECT || type == T_ARRAY) {
+      return !UseCompressedOops;
+    }
+#endif
```

Comment:
I'm confused why this is not needed today?

Reply:
> 

Today this is only used by typeArrayOops, afaict.